.github/workflows/cd.yml

```diff
@@ -203,6 +203,7 @@ jobs:
           platforms: ${{ matrix.arch.platform }}
           build-args: |
             PACKAGE=${{ matrix.name.package }}
+            API_URL=wss://api.firezone.dev
```

Comment:
@thomaseizinger @AndrewDryga

After much thought, I ended up going with using the prod URL here.

My thought process:

- Due to the complexity of our product (and Docker's lack of IPv6), `staging` is being used heavily to support product development -- local development of clients, relays, and gateways, is not really feasible (yet).
- Given the above, `staging` 

This is nice because it means our drafted artifacts will always target `prod`, which means the binaries we QA will be the same ones we publish, down to the sha256 checksum.

This is also nice because it mirrors the workflow we'd use to ship new cross-component product features (and loosely how we've been operating so far):

1. Build and ship necessary portal changes to `staging`
2. Rust/Client devs build against this env
3. Ship above changes to `prod` (perhaps with a feature flag / only enabled for the `firezone` account)
4. Test drafted clients/relays/gateways against `prod`
5. Publish drafted relay/clients/gateway -- by this point, the changes would have long been in the portal. Perhaps on the `publish` here we should only remove the feature flag.


Let's chat more at standup / this week -- I want to get your thoughts on this approach.

Reply:
cc @bmanifold for visibility too. Curious if anything jumps out here as a Bad Idea™

Reply:
> QA of drafted release artifacts should then ideally be done against prod (against our firezone account).

Just want to make sure I'm reading this statement correctly.  Are you saying our QA efforts will be done with the application environment set to `prod` or are you saying QA efforts will be done in our cloud `prod` environment?

Reply:
@bmanifold Moreso the latter. I should clarify what I mean by QA: any manual testing needed before "Go ship" for the data plane components. Once we publish clients/gateways/relays to Docker registries and App Stores we can't unpublish them, so I thought this would be our final Go or No go decision point.

For day to day building and manual testing, I think `staging` is working pretty well for us so far.

Reply:
TBH I'm not a big fan of those changes, let me go into details:

> This is also nice because it mirrors the workflow we'd use to ship new cross-component product features (and loosely how we've been operating so far):

This is just one of the workflows, IRL we will have all sorts of needs:

*Production bug in API*
1. Download and install the production binary;
2. Change its `API_URL` to localhost;
3. Reproduce and fix the bug;
4. Push fix to staging;
5. Change the app's `API_URL` to staging and make sure the issue is resolved.

*Production bug in mobile*
1. Download and install the production binary;
2. Change its `API_URL` to staging;
3. Fix the bug by making all sorts of needed requests (without polluting production) or by copying specific account settings from our clients production accounts to staging;
4. Change `API_URL` back to production to verify the issue is resolved.

*Backwards compatibility testing*
We should be able to test that the latest mobile client works with the current production version and that the latest staging API versions work with the oldest client (ideally released to production, so from App Store / Play Market).

> QA drafted clients/relays/gateways against prod

This basically makes our development cycles longer and riskier, because if mobile dev finds a bug or wants a slight change in a new feature (this will happen all the time) then we need a whole new cycle of very careful hot fixing and deploying to production, for something that nobody else will see, just because we risk taking the prod down.

If we can run everything on staging then we can move faster pushing whatever we need there and test across all the teams, having no constraints on our release cycles.

> to be configured at build-time, not runtime

Can we have a build-time configuration as `DEFAULT_API_URL` and the ability for a runtime override? Ideally, a feature on how to take a binary and override it locally.

Then we can have a separate **hidden** feature to change the `API_URL` in any released app (we can have both staging and production in our applinks or always redirect Android clients to the production applink).

> Enable the feature flag in the portal for all accounts

I don't think that we will have feature-flagging in the near future. It's a highly complex thing to maintain, especially for a small team. We will have billing enable/disable features, but not behavior changes controlled by flags.

Reply:
@AndrewDryga I think that makes sense. We discussed at standup / grooming this morning and all pretty much arrived at these same points.

I think the "Advanced settings" / _hidden feature_ solution solves our issue. I'll update this PR with those changes.

Reply:
Perfecto, thank you @jamilbk 👍